File: demo/assets/ydraw/lua/ymesh/gltf.lua
-- ydraw client interface target sketch — ymesh complex drawables (Lua).
-- NOT RUNNABLE YET — see python/ymesh/gltf.py. Assets: demo/assets/ymesh/.
local ydraw = require("yetty.ydraw")

local ASSETS = "demo/assets/ymesh/"

local function show(mesh)
    local dlist = ydraw.DrawableList()
    dlist:add(mesh)
    dlist:dcs_emit()
    dlist:destroy()
end

-- Default camera (frame-all), solid shading.
print("duck")
show(ydraw.Mesh{ASSETS .. "Duck.glb", width = 480, height = 360})

-- Camera posed via the same parameters the tool's orbit drag mutates.
print("avocado, posed camera")
show(ydraw.Mesh{ASSETS .. "Avocado.glb", width = 480, height = 360,
                azimuth = 0.6, elevation = 0.3, zoom = 1.4})

-- Wireframe toggle (the tool's W key).
print("box, wireframe")
show(ydraw.Mesh{ASSETS .. "Box.glb", width = 320, height = 240,
                wireframe = true})

File: demo/assets/ydraw/go/ymesh/gltf.go
// ydraw client interface target sketch — ymesh complex drawables (Go).
// NOT RUNNABLE YET — see python/ymesh/gltf.py. Assets: demo/assets/ymesh/.
package main

import (
	"fmt"

	"github.com/zokrezyl/yetty/bindings/go/ydraw"
)

const assets = "demo/assets/ymesh/"

func show(mesh ydraw.Mesh) {
	dlist := ydraw.NewDrawableList()
	dlist.Add(mesh)
	dlist.DcsEmit()
	dlist.Destroy()
}

func main() {
	// Default camera (frame-all), solid shading.
	fmt.Println("duck")
	show(ydraw.Mesh{Path: assets + "Duck.glb", Width: 480, Height: 360})

	// Camera posed via the same parameters the tool's orbit drag mutates.
	fmt.Println("avocado, posed camera")
	show(ydraw.Mesh{Path: assets + "Avocado.glb", Width: 480, Height: 360,
		Azimuth: 0.6, Elevation: 0.3, Zoom: 1.4})

	// Wireframe toggle (the tool's W key).
	fmt.Println("box, wireframe")
	show(ydraw.Mesh{Path: assets + "Box.glb", Width: 320, Height: 240,
		Wireframe: true})
}
